pyslackersweb/contexts.py

```diff
@@ -0,0 +1,53 @@
+from typing import AsyncGenerator
+
+import aioredis
+from aiohttp import ClientSession, web
+from apscheduler.schedulers.asyncio import AsyncIOScheduler
+
+
+def _register_for_subapps(app, key, obj):
```

Comment:
@ovv I am not sure I want this or not, one idea was to inspect the nested application to see if it has the matching key registered as None we should attach...

Still thinking through this.

Reply:
I think I am going another direction with this, I think it makes sense to just do the connection opening/setup in the top level `app_factory` and register signals for any added lifecycle methods we need to hit (such as start/stop for apscheduler, wait on redis to close, etc).